nwbwidgets/misc.py

```diff
@@ -106,7 +106,11 @@ def show_session_raster(
         progress_bar=progress_bar,
     )
     ax.set_ylabel("unit #")
-    ax.set_yticklabels(units.id.data[:])
+    if len(data) <= 30:
+        unit_id_display = np.array(units.id.data[:])[[x for x in this_iter]]
```

Comment:
The previous version did not actually filter the unit ids based on the unit selector bar - this version does.